src/user/event_service.rst

```diff
@@ -0,0 +1,141 @@
+  to accomplish. A workfow comprises a group of constituent tasks and their topological relationship.
+
+- **Task**. A task is a software program or application that utilizes computing resources to achieve one
+  of the intermediate steps in a workfow. It normally processes a set of data.
+
+- **Job**. A job is a PanDA object with distinct states that corresponds a sub-unit of a task. A single
+  task consists of multiple jobs, and each job is executed by a worker, which is an abstraction of the
+  execution point, such as a worker node and CPU cluster. A job can process a list of files, a file or
+  a part of one file. In ATLAS experiment, one file normally stores many ordered events (particle objects).
+  One job can process a part of the jobs, for example 0~1000 events for the first job and 1000~2000 for
+  another job. In the job level, the events are processed in order in one sequence. If one of the fails,
+  the job will fail.
+
+- **EventService**. EventService introduces another bookkeeping level which split a job into a fine-grained
+  processing level. With EventService, the events in a file can be processed discretely. The bookkeeping
+  and retrying policy can be based on events.
+
+Overheads in Job Scheduling
+---------------------------
+
+During job scheduling, there are overheads at many places. If the payload is too small, the overhead
+may be a big part from the payload submitted to the output returned back to users. It's good to make
+sure the payload not too small.
+
+- **scheduling period**. The basic scheduling loop is normally similar to the script described below.
+  Normally the scheduler should sleep a period between different scheduling loops. With a sleep period,
+  it will make all system busy and not efficient. If checking the configuration of workflow/workload
+  system such as Condor, Slurm, BPS, ARC CE, PanDA, Harvester and iDDS, you will find the interval
+  period defined in different systems. By default, the interval period normally can be 30 seconds
+  or 1 minute. When the system is big with many resources and users, the interval period can be even
+  bigger. At the same time, frequently there are multiple agents or threads in one system. If one job
+  involves many systems, such as iDDS->PanDA->Harvester->ARC CE->Cluster system like Slurm, and many
+  agents/threads in one system, the total overhead can be a few minutes. ::
+
+    while not terminated:
+      schedule_jobs()
+      sleep(interval_period)
+
+- **Processing overhead**. For a payload, when it begins, it needs some steps to setup the environment.
+  When it ends, it needs some time to transfer the outputs/logs to some storages. It also takes some time.
+
+EventService
+------------
+
+How does EventService run:
+
+- In EventService, events in a file is attached to a job or multiple jobs.
+
+- When pilot starts EventService, here is how does a job process multiple events::
+
+    (pilot)
+    get_job
+    setup_job
+    while more_events:
```

Comment:
while more_events_for_job:  
(i.e., the more events is limited to this job, right?)